From 7d75d97ff1d071bc688d74f7f5060277ee82c743 Mon Sep 17 00:00:00 2001
From: Sebastian Bazley <sebb@apache.org>
Date: Mon, 7 Apr 2008 09:55:11 +0000
Subject: [PATCH] Update setLenient tests: - check shortdates work with lenient
 true or false for non-future dates - add test for NET-211

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/net/trunk@645435 13f79535-47bb-0310-9956-ffa450edef68
---
 .../net/ftp/parser/FTPTimestampParserImplTest.java  | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/test/org/apache/commons/net/ftp/parser/FTPTimestampParserImplTest.java b/src/test/org/apache/commons/net/ftp/parser/FTPTimestampParserImplTest.java
index a0bae3c70..b32296ec2 100644
--- a/src/test/org/apache/commons/net/ftp/parser/FTPTimestampParserImplTest.java
+++ b/src/test/org/apache/commons/net/ftp/parser/FTPTimestampParserImplTest.java
@@ -224,14 +224,17 @@ public void testParser() {
 	}
 	
     /*
-     * Check how short date is interpreted at a given time
+     * Check how short date is interpreted at a given time.
+     * Check both with and without lenient future dates
      */
     private void checkShortParse(String msg, Calendar now, Calendar input) throws ParseException {
         checkShortParse(msg, now, input, false);
+        checkShortParse(msg, now, input, true);
     }
 
     /*
      * Check how short date is interpreted at a given time
+     * Check only using specified lenient future dates setting
      */
     private void checkShortParse(String msg, Calendar now, Calendar input, boolean lenient) throws ParseException {
         FTPTimestampParserImpl parser = new FTPTimestampParserImpl();
@@ -351,6 +354,14 @@ public void testParseJan01Lenient() throws Exception {
         checkShortParse("2008-1-1",now,target, true);
     }
 
+    public void testParseJan01() throws Exception {
+        GregorianCalendar now = new GregorianCalendar(2007, Calendar.JANUARY, 1, 12, 0);
+        checkShortParse("2007-01-01",now,now); // should always work
+        GregorianCalendar target = new GregorianCalendar(2006, Calendar.DECEMBER, 31, 12, 0);
+        checkShortParse("2006-12-31",now,target, true);
+        checkShortParse("2006-12-31",now,target, false);
+    }
+
     /**
      * Method suite.
      *
